specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json

```diff
@@ -691,72 +691,6 @@
         ]
       }
     },
-    "/subscriptions/{subscriptionId}/resources": {
```

Comment:
This breaks a lot of scripts, including ones we use to monitor cost analysis for the Azure SDK. Why is listing vaults in a sub being removed? Could you comment here, or even update the PR description with reasoning? This is a pretty big change.

Reply:
Hi @heaths , the actual key vault list API is on line 470: “/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults”
The removing API seems doesn’t belong to us. When customer call GET with the removing url: "/subscriptions/{subscriptionId}/resources", it goes to another service endpoint, which I believed it is here: [Microsoft.Resource](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2021-01-01/resources.json#L2881). So this make us unable to control this API as well as its version. I am not totally sure if I am wrong here and I don’t know why this API is listed in our code for a long time.

Reply:
It [was in 2019-09-01](https://github.com/Azure/azure-rest-api-specs/blob/5b1c5d06c84680cfc4c5a179c56ec8b1d9a0e295/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/keyvault.json#L694) GA, so this is breaking. But is that desirable in this case? I worry about clients already relying on it.

Reply:
It was in all the previous GA versions as well. But I am not sure if previous SDKs already have a way to due with it. I talked with team before. Looks like it is desirable to have this ARM API in our spec.